### PR TITLE
first try solution

### PR DESCRIPTION
**Pull request by AndreyDMA**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,14 +3,38 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K,V> {
+    int count = 0;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: add access modifier

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,14 +3,38 @@
...
     @Override
     public V get(K key) {
+        for (int i = 0; i < count; i++) {
+            if (key == null || key.equals(keyArray[i])) {
+                return (V) valueArray[i];
+            }
+        }
         return null;
     }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: remove redundant lines

Comment (AndreyDMA) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: fixed

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,13 +3,33 @@
...
+    private Object[] valueArray = new Object[16];
 
     @Override
     public void put(K key, V value) {
+        if (count == 0) {
+            keyArray[count] = key;
+            valueArray[count] = value;
+        }
+        for (int i = 0; i < count; i++) {
+            if (keyArray[i].equals(key)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: how will work this method if key is null?

Comment (AndreyDMA) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: added null check...

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,13 +3,33 @@
...
+    private Object[] valueArray = new Object[16];
 
     @Override
     public void put(K key, V value) {
+        if (count == 0) {
+            keyArray[count] = key;
+            valueArray[count] = value;
+        }
+        for (int i = 0; i < count; i++) {
+            if (key != null && keyArray[i].equals(key)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]:
```suggestion
            if (key == null && keyArray[i]==null || keyArray[i].equals(key)) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,13 +3,33 @@
...
+            keyArray[count] = key;
+            valueArray[count] = value;
+        }
+        count++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < count; i++) {
+            if (key == null || key.equals(keyArray[i])) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: the same

Comment (AndreyDMA) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: I have to add exception? or just not allow to use null key?

Comment (AndreyDMA) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Thanks for hints)

**Pull request closed without merging**